### ISSUE TITLE
Method for cropping geoarrays

### ISSUE THREAD

**Issue by Lucyhenley** [evetion/GeoArrays.jl#39]
It would be useful to have some method that would crop a geoarray to a given extent, possibly something like this R function: 
https://www.rdocumentation.org/packages/raster/versions/3.0-12/topics/crop

It takes either coordinates of the box to be cropped to or a separate raster with the desired extent.

**Comment by evetion** [comment]
That seems a fair use case. This would then be:

`crop!(ga::GeoArray, bbox::NamedTuple)`
And
`crop!(ga::GeoArray, ga_to_crop_with::GeoArray)`

For the last signature, it can only work for GeoArrays with the same CRS. It will throw a warning if one of the rasters has a different affine transform (other resolution/rotation), because the cropping will introduce errors.

**Comment by KrOstir** [comment]
Cropping have been implemented in #78.

```julia
function crop(ga::GeoArray, cbox::NamedTuple{(:min_x, :min_y, :max_x, :max_y)})
function crop(ga::GeoArray, cga::GeoArray)
```

The first method uses a cropping box, the second method crops by another GeoArray.
